src/views/projects/SourceBrowser/FileDiff.svelte

```diff
@@ -16,6 +17,53 @@
   export let projectId: string;
 
   let collapsed = false;
+  let selection: Selection | undefined = undefined;
+
+  onMount(() => {
+    window.addEventListener("keydown", e => {
+      if (e.key === "Escape") {
+        updateHash("");
+      }
+    });
+    window.addEventListener("hashchange", updateSelection);
+
+    return () => window.removeEventListener("hashchange", updateSelection);
+  });
+
+  if (file.diff.type === "plain" && file.diff.hunks.length > 0) {
+    updateSelection();
+    for (const hunkIdx in file.diff.hunks) {
+      const hunk = file.diff.hunks[hunkIdx];
+      for (const lineIdx in hunk.lines) {
+        if (shouldScroll(hunkIdx, lineIdx)) {
+          setTimeout(() => {
+            const target = document.getElementById(
+              file.path + "::H" + hunkIdx + "L" + lineIdx,
+            );
+            if (target) {
+              target.scrollIntoView();
+            }
+          }, 200);
+        }
+      }
+    }
+  }
```

Comment:
You can replace all of this code with the following:

```suggestion
  onMount(() => {
    updateSelection();

    if (selection) {
      document
        .getElementById(
          file.path + "::H" + selection.startHunk + "L" + selection.startLine,
        )
        ?.scrollIntoView();
    }
  });
```

There’s no need for the the timeout because we are running after mounting.

I would also suggest to simplify the ID format and not rely on a subtle difference to the hash format to prevent scrolling. Something like `[file.path, selection.startHunk, selection.startLine].join("-")`

Reply:
https://github.com/radicle-dev/radicle-interface/pull/942/commits/1b19c1999d0a7f99a2cf91bbef7a076d7be2594e